src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/GenericCompositeBindingProvider.cs

```diff
@@ -67,7 +68,19 @@ public GenericCompositeBindingProvider(IEnumerable<IBindingProvider> providers)
             {
                 resourceName = resourceName.Substring(0, resourceName.Length - Suffix.Length);
             }
-            throw new InvalidOperationException("Can't bind " + resourceName + " to type '" + context.Parameter.ParameterType + "'.");
+
+            StringBuilder exceptionMessage = new StringBuilder($"Can't bind {resourceName} to type '{context.Parameter.ParameterType}'.");
+
+            if (context.BindingErrors.Count > 0)
+            {
+                exceptionMessage.AppendLine().AppendLine("Possible causes:");
+                foreach (var error in context.BindingErrors.Select((value, index) => new { Value = value, Index = index + 1 }))
```

Comment:
>Select((value, index) => new { Value = value, Index = index + 1 }) [](start = 60, length = 66)

nit: can we avoid this extra select clause and just use a local? 
Ie, something like: 
    var x = index+1
    $"{x}) {error.value}"

Reply:
Sure; I thought the numbering improved readability, but I can also remove it entirely if preferred.